src/main/java/be/seeseemelk/mockbukkit/block/state/SignMock.java

```diff
@@ -120,10 +120,9 @@ public void line(int index, @NotNull Component line) throws IndexOutOfBoundsExce
 
 	@Override
 	@Deprecated
-	public void setLine(int index, @NotNull String line) throws IndexOutOfBoundsException
+	public void setLine(int index, String line) throws IndexOutOfBoundsException
```

Comment:
While the Implementation doesn't directly enforce Nullability here, we should keep the Annotation here, sine this would be inconsistent with upstream Spigot

Reply:
If I'm correct (I'm not exactly sure of the changes of the paper API, if any), this will be a conflict of behavior between the spigot and the paper API; paper would not have the NotNull annotation while spigot will.

I can also confirm that the implementation in spigot allows for null sign lines.

I'm unsure on what would be best here though.

Reply:
As can be seen [here](https://jd.papermc.io/paper/1.20/org/bukkit/block/Sign.html#setLine(int,java.lang.String)). Paper still has the Annotation. The only change to the paper API is that [org.bukkit.block.Sign#line(int,net.kyori.adventure.text.Component)](https://jd.papermc.io/paper/1.20/org/bukkit/block/Sign.html#line(int,net.kyori.adventure.text.Component)) now enforces the Nullability with a Precondition

Reply:
Okay, noted. Then definitely